src/impl_methods.rs

```diff
@@ -1911,13 +1933,68 @@ where
         // Check if contiguous, if not => copy all, else just adapt strides
         unsafe {
             // safe because arrays are contiguous and len is unchanged
-            if self.is_standard_layout() {
-                Ok(self.with_strides_dim(shape.default_strides(), shape))
-            } else if self.ndim() > 1 && self.raw_view().reversed_axes().is_standard_layout() {
-                Ok(self.with_strides_dim(shape.fortran_strides(), shape))
-            } else {
-                Err(error::from_kind(error::ErrorKind::IncompatibleLayout))
+            match order {
+                None | Some(Order::RowMajor) if self.is_standard_layout() => {
+                    Ok(self.with_strides_dim(shape.default_strides(), shape))
+                }
+                None | Some(Order::ColumnMajor) if self.raw_view().reversed_axes().is_standard_layout() => {
+                    Ok(self.with_strides_dim(shape.fortran_strides(), shape))
+                }
+                _otherwise => Err(error::from_kind(error::ErrorKind::IncompatibleLayout))
+            }
+        }
+    }
+
+    pub fn into_shape_clone<E>(self, shape: E) -> Result<ArrayBase<S, E::Dim>, ShapeError>
```

Comment:
Into shape with clone fallback has better usability. But it doesn't work for the view usecase